docs/content/014-guides/022-source-types.mdx

```diff
@@ -0,0 +1,449 @@
+      //                                      Make sure to use unique names if you have multiple sources
+    }]
+  }
+})
+```
+
+In the example above, since the source type `interface User` matches the name of the `User` GraphQL Object type, Nexus will use the `interface User` as source type for the `User` GraphQL Object type.
+
+#### Different names following a convention
+
+It is however possible that your source types don't exactly match the name of your GraphQL Object type but follow a convention.
+
+For instance:
+
+- source type `class UserModel {}` -> GraphQL Object type `User`
+- source type `class PostModel {}` -> GraphQL Object type `Post`
+- source type `class CommentModel {}` -> GraphQL Object type `Comment`
+
+In that case, you can configure your `source` to teach Nexus about your convention using regexes:
+
+```ts
+//db.ts
+
+export class UserModel {
+// ^-------------^---------- Export your source type (required)
+//               |---------- Name of your source type which will be extracted by Nexus
+  firstName: string
+  lastName: string
+}
+
+export class PostModel {
+// ^-------------^---------- Export your source type (required)
+//               |---------- Name of your source type which will be extracted by Nexus
+// ...
+}
+
+export class CommentModel {
+// ^-------------^---------- Export your source type (required)
+//               |---------- Name of your source type which will be extracted by Nexus
+// ...
+}
+
+// schema.ts
+import { makeSchema, objectType, queryType } from '@nexus/schema'
+import * as path from 'path'
+
+const User = objectType({
+  name: 'User',
+  //      ^---------------------- Name of your GraphQL Object type which will be matched against the name of your source types
+  definition(t) {
+    t.string('fullName', {
+      resolve(user) {
+        //      ^------------------------ as: UserModel
+        return user.firstName + ' ' + user.lastName
+      }
+    })
+  }
+})
+
+const Query = queryType({
+  definition(t) {
+    t.list.field('users', {
+      type: User,
+      resolve(_parent, args, ctx) {
+        //                       ^------- return as: UserModel[]
+        // ...
+      }
+    })
+  }
+})
+
+const schema = makeSchema({
+  types: [User, Query],
+  typegenAutoConfig: {
+    sources: [{
+      source: path.join(__dirname, 'db.ts'),
+      //       ^----------------------------- Path to the file containing your source types
+      alias: 'db',
+      //      ^------------------------------ Name used as alias for Nexus to import your source types. eg: import * as <name> from <path>
+      typeMatch(type, defaultRegex) {
+      //         ^--------^---------------- GraphQL Named type to match against your source types
+      //                  |---------------- Default regex used to match your GraphQL types against your source types
+        return new RegExp(`(?:interface|type|class|enum)\\s+(${type.name}Model)\\W`, 'g')
```

Comment:
It took me a while to realize this is regex run literally against source code syntax. I think we need to make that a bit clearer here? 

Probably in a few places.

For example above:

>  to teach Nexus about your convention using regexes

Maybe:


> to teach Nexus about your convention. The way it works is that you give regular expressions to Nexus which it will run against the literal source code within modules (also specified by you) to find which types should be used for source types. Your regular expression must return a regex capturing group containing the TS type name of the source type. Here is a complete and annotated example to illustrate the system:


ish?

Reply:
![image](https://user-images.githubusercontent.com/25233379/101619104-15d4f680-3a13-11eb-8105-30bf3a32c631.png)

Thanks for suggestion 🙏